controller/backing_image_controller.go

```diff
@@ -503,37 +485,26 @@ func (bic *BackingImageController) handleBackingImageDataSource(bi *longhorn.Bac
 		case longhorn.BackingImageDataSourceTypeDownload:
 			log.Info("Prepare to re-download backing image via backing image data source since all existing files become unavailable")
 			bids.Spec.FileTransferred = false
+			bids.Spec.NodeID = ""
+			bids.Spec.DiskUUID = ""
+			bids.Spec.DiskPath = ""
 		default:
 			log.Warnf("Cannot recover backing image after all existing files becoming unavailable, since the backing image data source with type %v doesn't support restarting", bids.Spec.SourceType)
 		}
 	}
 
 	if !bids.Spec.FileTransferred {
-		if _, _, err := bic.ds.GetReadyDiskNode(bids.Spec.DiskUUID); err != nil && types.ErrorIsNotFound(err) {
-			nodes, err := bic.ds.ListReadyNodes()
-			if err != nil {
-				return err
-			}
-			for _, node := range nodes {
-				updated := false
-				for diskName, diskSpec := range node.Spec.Disks {
-					diskStatus, exists := node.Status.DiskStatus[diskName]
-					if !exists {
-						continue
-					}
-					if types.GetCondition(diskStatus.Conditions, longhorn.DiskConditionTypeSchedulable).Status != longhorn.ConditionStatusTrue {
-						continue
-					}
-					bids.Spec.DiskUUID = diskStatus.DiskUUID
-					bids.Spec.DiskPath = diskSpec.Path
-					bids.Spec.NodeID = node.Name
-					updated = true
-				}
-				if updated {
-					break
-				}
-			}
+		node, diskName, err := bic.ds.GetReadyDiskNode(bids.Spec.DiskUUID)
```

Comment:
Why not just use `bids.Status.CurrentState`and `bids.Spec.FileTransferred` together to decide if need to update the `bids.Spec.NodeID/Disk..`?

Reply:
Double-checked the logic, we don't need to validate `bids.Status.CurrentState`. See the comment in the code for details.

Reply:
Yeah, this can be improved later. If needed, create a ticket for this.